src/lib/homestore/heap_chunk_selector.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <homestore/chunk_selector.h>
+#include <homestore/vchunk.h>
+#include <homestore/homestore_decl.hpp>
+
+#include <queue>
+#include <vector>
+#include <mutex>
+#include <utility>
+#include <memory>
+
+#include <folly/concurrency/ConcurrentHashMap.h>
+
+namespace homeobject {
+
+using csharedChunk = homestore::cshared<homestore::Chunk>;  
+
+class HeapChunkSelector : public homestore::ChunkSelector {
+public:
+    HeapChunkSelector() = default;
+    HeapChunkSelector(const HeapChunkSelector&) = delete;
+    HeapChunkSelector(HeapChunkSelector&&) noexcept = delete;
+    HeapChunkSelector& operator=(const HeapChunkSelector&) = delete;
+    HeapChunkSelector& operator=(HeapChunkSelector&&) noexcept = delete;
+    ~HeapChunkSelector() = default;
+
+    using VChunk = homestore::VChunk;
+
+    class VChunkComparator {
+    public:
+        bool operator()(VChunk& lhs, VChunk& rhs) {
+            return lhs.available_blks() < rhs.available_blks();
+        }
+    };
+
+    using VChunkHeap = std::priority_queue<VChunk, std::vector<VChunk>, VChunkComparator>;
+
+    void add_chunk(csharedChunk&) override;
```

Comment:
I realize this is from HomeStore's interface but should be either:
`void add_chunk(csharedChunk const&) override; // < Hari's Preference`
or
`void add_chunk(csharedChunk) override; // < Brian's Preference`
But we intend to take ownership of the chunk here yeah, not mutate the caller's own stack.

Reply:
yes , we do not want to mutate chunk in the caller`s stack , will change this in the homestore pr

Reply:
So; I misunderstood the `c` effecting the value type, not being simply an alias for `const shared<>`; I'm okay with the signature, but the alias i'd prefer to just see:
```
void add_chunk(const sharedChunk&) override;
```
but it's a nitpick now that i understand this alias.